c/tests/test_minimal_cpp.cpp

```diff
@@ -78,12 +80,163 @@ test_table_basics()
     tsk_table_collection_free(&tables);
 }
 
+/* A definition of sort_edges that uses C++ std::sort and inlining of the
+ * comparison function to achieve significantly better performance than
+ * the builtin method in tskit.
+ */
+int
+cpp_sort_edges(tsk_table_sorter_t *sorter, tsk_size_t start)
+{
+    struct _edge {
+        double left, right;
+        tsk_id_t parent, child;
+
+        _edge(double l, double r, tsk_id_t p, tsk_id_t c)
+            : left{ l }, right{ r }, parent{ p }, child{ c }
+        {
+        }
+    };
+    tsk_edge_table_t *edges = &sorter->tables->edges;
+    const double *node_time = sorter->tables->nodes.time;
+    std::vector<_edge> sorted_edges;
+    size_t num_edges = edges->num_rows;
+    size_t j;
+
+    /* This is the comparison function.  We cannot define an
+     * operator < for _edge because we need to bind the node times
```

Comment:
I didn't try out that approach.  Part of the difference is that the fwdpp tables are AoS layout, as opposed to tskit's SoA layout, so fwdpp never makes this copy to an intermediate to begin with, so copying the times doesn't come up.  I went with an AoS layout precisely so that `std::sort` could be used.

Reply:
I see.  Interesting; I didn't realize fwdpp rolled its own table stuff to such an extent.  Well, I'll try the performance comparison in SLiM.

Reply:
The performance results that I presented to the group used this same lambda structure, FWIW.

Reply:
> I see. Interesting; I didn't realize fwdpp rolled its own table stuff to such an extent. 

This is how I really dug in to understand the approach.  The mutation table is different, too, and can be used for garbage-collecting extinct variants.  Other tables I don't use, as they're totally redundant with other concepts and only need to be generated in Python at the end of a fwdpy11 simulation if a `tskit` object is requested.

Reply:
> I see. Interesting; I didn't realize fwdpp rolled its own table stuff to such an extent. Well, I'll try the performance comparison in SLiM.

One argument for not copying the times over is that you can reduce the copy from 2 `int32_t` + 3 `double` down to 2 `int32_t` and 2 `double`, which is 25% less.

Reply:
> One argument for not copying the times over is that you can reduce the copy from 2 `int32_t` + 3 `double` down to 2 `int32_t` and 2 `double`, which is 25% less.

Yes; so you avoid N double-size copies, for sure.  But now you do N log N comparisons, each of which has to do two extra lookups to get the times, and those times are also in a different memory block so there's less locality.  I dunno, but it feels like doing the copy ought to be faster.  Anyway, I'll try it.  :->

Reply:
I think Ben is probably right here, locality and cache friendliness will win out. It'll be fun to try it out on the example program!